### ISSUE TITLE
TPipeline can't figure out right part of statusline with feline.nvim

### ISSUE THREAD

**Issue by primeapple** [vimpostor/vim-tpipeline#60]
### Describe the bug

While working on a deeper integration of the kitty statusline with neovim via this plugin, I encountered a problem.

Apparently tpipeline can't figure out the left and right part of my statusline, when using https://github.com/freddiehaddad/feline.nvim . The file for the right status line is always empty and everything is in the left statusline file.

Maybe it has something to do with me not runnig tpipeline in tmux? Or I think the problem is that tpipeline can't figure out the separator between left and right status config. Can you tell me how you try to find left and right?

### To reproduce

My feline and tpipeline config can be found here: 

https://github.com/primeapple/dotfiles/blob/main/.config/nvim/lua/plugins/statusline.lua

My simplified tab_bar.py is here (but is not important for this bug) https://github.com/primeapple/dotfiles/blob/main/.config/kitty/tab_bar.py

### Terminal emulator

Kitty

### Debug output

```Text
{'native_str': ' o///h/t/.c/n/█                                                                                    █O
IL██1:10██Top█▇▇', 'brand': 'neovim', 'right': [''], 'job_state': 'run as PID 9123', 'bad_colors': 9, 'job_errors': []
, 'tpipeline_size': 0, 'version_info': '{   api_compatible = 0,   api_level = 11,   api_prerelease = false,   major = 
0,   minor = 9,   patch = 5,   prerelease = false }', 'tmux': '^~~^', 'left': ['#[fg=#c8c093,bg=#16161d]#[fg=#6d8086,b
g=#1f1f28,bold]#[fg=#dcd7ba,bg=#1f1f28,bold] o///h/t/.c/n/#[fg=#1f1f28,bg=#1f1f28,nobold]█#[fg=#938aa9,bg=#1f1f28]#[f
g=#76946a,bg=#1f1f28]#[fg=#c34043,bg=#1f1f28]#[fg=#dca561,bg=#1f1f28]#[fg=#dcd7ba,bg=#1f1f28]#[fg=#938aa9,bg=#1f1f28,b
old]#[fg=#e82424,bg=#1f1f28,nobold]#[fg=#ff9e3b,bg=#1f1f28]#[fg=#658594,bg=#1f1f28]#[fg=#6a9589,bg=#1f1f28]#[fg=#1f1f2
8,bg=#1f1f28]█#[fg=#dcd7ba,bg=#1f1f28,bold]OIL#[fg=#1f1f28,bg=#1f1f28,nobold]█#[fg=#1f1f28,bg=#1f1f28]█#[fg=#dcd7ba,bg
=#1f1f28,bold]1:10#[fg=#1f1f28,bg=#1f1f28,nobold]█#[fg=#1f1f28,bg=#1f1f28]█#[fg=#dcd7ba,bg=#1f1f28,bold]Top#[fg=#1f1f2
8,bg=#1f1f28,nobold]█#[fg=#1f1f28,bg=#dcd7ba,bold]▇▇'], 'plugin_version': '1.0', 'native_highlights': [{'group': 'Stat
usComponent_6d8086_1F1F28_bold', 'start': 0}, {'group': 'StatusComponent_DCD7BA_1F1F28_bold', 'start': 3}, {'group': '
StatusComponent_1F1F28_1F1F28_NONE', 'start': 17}, {'group': 'StatusComponent_938AA9_1F1F28_NONE', 'start': 20}, {'gro
up': 'StatusComponent_76946A_1F1F28_NONE', 'start': 20}, {'group': 'StatusComponent_C34043_1F1F28_NONE', 'start': 20},
 {'group': 'StatusComponent_DCA561_1F1F28_NONE', 'start': 20}, {'group': 'StatusComponent_DCD7BA_1F1F28_NONE', 'start'
: 20}, {'group': 'StatusComponent_938AA9_1F1F28_bold', 'start': 104}, {'group': 'StatusComponent_E82424_1F1F28_NONE', 
'start': 104}, {'group': 'StatusComponent_FF9E3B_1F1F28_NONE', 'start': 104}, {'group': 'StatusComponent_658594_1F1F28
_NONE', 'start': 104}, {'group': 'StatusComponent_6A9589_1F1F28_NONE', 'start': 104}, {'group': 'StatusComponent_1F1F2
8_1F1F28_NONE', 'start': 104}, {'group': 'StatusComponent_DCD7BA_1F1F28_bold', 'start': 107}, {'group': 'StatusCompone
nt_1F1F28_1F1F28_NONE', 'start': 110}, {'group': 'StatusComponent_1F1F28_1F1F28_NONE', 'start': 113}, {'group': 'Statu
sComponent_DCD7BA_1F1F28_bold', 'start': 116}, {'group': 'StatusComponent_1F1F28_1F1F28_NONE', 'start': 120}, {'group'
: 'StatusComponent_1F1F28_1F1F28_NONE', 'start': 123}, {'group': 'StatusComponent_DCD7BA_1F1F28_bold', 'start': 126}, 
{'group': 'StatusComponent_1F1F28_1F1F28_NONE', 'start': 129}, {'group': 'StatusComponent_1F1F28_DCD7BA_bold', 'start'
: 132}], 'os': 'Linux', 'stl': '%{%v:lua.require''feline''.generate_statusline()%}'}
```

**Comment by vimpostor** [comment]
> 'tpipeline_size': 0

This plugin tries to find out the size of tmux (which is `0` because you are not using tmux), which causes your mentioned problem. Fixing this is trivial, but it might require the user to provide a custom autocmd to set a value for `g:tpipeline_size`. In your case it is probably as easy as running `let g:tpipeline_size = &co`.

As a quick test, can you check if the above `:let g:tpipeline_size = &co` and `:au VimResized * let g:tpipeline_size = &co` fixes the problem for you?

I still need to think about how to best fix this for the general case, so that it can work with all external statusline integrations, `&co` might not be the correct value if the statusline is embedded in the window manager status line for example and it will also fail if you are using kitty split windows.

> My simplified tab_bar.py is here (but is not important for this bug) https://github.com/primeapple/dotfiles/blob/main/.config/kitty/tab_bar.py

Interesting, that you are writing a custom script, did you not see the already existing solution in `:h tpipeline-kitty`?

**Comment by vimpostor** [comment]
It should work now for your statusline, if you use something like `autocmd User TpipelineSize let g:tpipeline_size = &co`.

**Comment by primeapple** [comment]
Thanks, your suggestion did help. I think this may require a fix within the plugin? So set `g:tpipeline_size = &co` whenever vim is not in tmux?

> Interesting, that you are writing a custom script, did you not see the already existing solution in :h tpipeline-kitty?

I saw it (in fact I was the guy that requested it in the first place 1,5 years ago ;-)). But I wanted to improve on that since it's very basic. My goal right now is to include the kitty tabs in the middle of the statusline. Also there are some bugs in the script (for example the default background is not handled properly).

I have a question, tho. Could you give me the datatype for the string in the status line file? 

I think it is something like (in TypeScript syntax):
```
// each element looks like this
#[fg=colorhex,bg=colorhex,???]text
```
Is `colorhex` always a hex? When is it the string `DEFAULT`? What values can be in `???`? Is there a fourth part that can be there? Did I miss anything?

**Comment by vimpostor** [comment]
> Also there are some bugs in the script (for example the default background is not handled properly).

That should already be fixed with 7d1fd627fc7e611fc3ba48a76e6f16b05a8b63ec

> I have a question, tho. Could you give me the datatype for the string in the status line file?
> ```
> // each element looks like this
> #[fg=colorhex,bg=colorhex,???]text
> ```

The string is formatted according to the tmux style rules, they are explained in detail in the `STYLES` section in the `tmux(1)` man page:
```
fg=colour
Set  the  foreground  colour.   The  colour is one of: default for the default colour; terminal for the terminal default colour; or a hexadecimal RGB string such as ‘#ffffff’.

bg=colour
Set the background colour.
```

**Comment by primeapple** [comment]
Just in case you are interested, here is a screenshot on what I have figured out: https://github.com/kovidgoyal/kitty/discussions/4447#discussioncomment-8575542
